### PR TITLE
Add insertion sort functions for Scala

### PR DESCRIPTION
**Pull request by varunganesh91**

### CODE REVIEW

**Comment by nicholette** [comment]
Can I nominate @raymondtay to review this PR? 😂

File: scala/sort/insertion_sort.sc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+     * Sorts the given list using the iterative approach to insertion sort
+     * 
+     * @param toSort the list of integers that needs to be sorted
+     * @param sortDesc sorts in descending order
+     */
+    def insertionSortImperative(toSort: List[Int], sortDesc: Boolean = false): List[Int] = {
+        // This logic requires a mutable alternative to the list
+        var arr = toSort.toArray
+        for (i <- 1 until arr.length) {
+            var j = i - 1
```
Comment (Bertrand31) [inline scala/sort/insertion_sort.sc]: OOooohh my eyes!!! Mutation! Vade retro! @varunganesh91

**Comment by tikal** [comment]
@Bertrand31 @varunganesh91 
Why don't we have test ? I mean they would both help explain the expectations and also help to develop, especially using `~testQuick` with sbt

**Comment by Bertrand31** [comment]
@tikal cf. https://github.com/nugit/LYANL/pull/3#issuecomment-428073542

File: scala/sort/insertion_sort.sc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+object InsertionSort {
+    private val notInOrder = (x1: Int, x2: Int, desc: Boolean) => {
```
Comment (nicholette) [inline scala/sort/insertion_sort.sc]: Sorry for the late review, it will be cleaner if we don't work with negations in method names 😅

**Pull request merged**